### PR TITLE
feat: [PL-40677]: add enable/disable notification channel

### PR DESCRIPTION
**Pull request by namanharness**
Adding Notification Channels for `Pager Duty`, `Webhook`, `Slack` & `MS Teams` to Account Settings 

###Feature:

https://github.com/harness/harness-core-ui/assets/110456765/95bc0f31-adce-4225-a086-7df97550925b

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 1]
```diff
@@ -99,6 +99,28 @@ DefaultSettingsFactory.registerCategory('EULA', {
   ]
 })
 
+DefaultSettingsFactory.registerCategory('NOTIFICATIONS', {
```
Comment (samarth-sharma31) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:102]: Lets use SettingGroups enum instead of using the 'NOTIFICATIONS' directly

Comment (pankajsingh-harness) [reply src/modules/27-platform/default-settings/RouteDestinations.tsx:102]: @samarth-sharma31 this is fine. 
```
export type SettingCategory = SettingDTO['category']

export interface SettingDTO {
  allowOverrides: boolean
  allowedScopes: ('ACCOUNT' | 'ORGANIZATION' | 'PROJECT')[]
  allowedValues?: string[]
  category:
    | 'CD'
    | 'CI'
    | 'CE'
    | 'CV'
    | 'CF'
    | 'STO'
    | 'CORE'
    | 'PMS'
    | 'TEMPLATESERVICE'
    | 'GOVERNANCE'
    | 'CHAOS'
    | 'SCIM'
    | 'GIT_EXPERIENCE'
    | 'CONNECTORS'
    | 'EULA'
    | 'NOTIFICATIONS'
  defaultValue?: string
  groupIdentifier: string
  identifier: string
  isSettingEditable: boolean
  name: string
  orgIdentifier?: string
  projectIdentifier?: string
  settingSource?: 'ACCOUNT' | 'ORG' | 'PROJECT' | 'DEFAULT'
  value?: string
  valueType: 'String' | 'Boolean' | 'Number'
}

```

File: src/modules/10-common/constants/Utils.ts [hunk 2]
```diff
@@ -112,7 +112,11 @@ export enum SettingType {
   EXPORT_SERVICE_VARIABLES_AS_ENV_VARIABLES = 'export_service_variables_as_env_variables',
   TICKETING_TOOL = 'ticketing_tool',
   TICKETING_TOOL_CONNECTOR = 'ticketing_tool_connector',
-  AIDA = 'aida'
+  AIDA = 'aida',
+  ENABLE_PAGERDUTY_NOTIFICATION_IDENTIFIER = 'enable_pagerduty_notification',
+  ENABLE_WEBHOOK_NOTIFICATION_IDENTIFIER = 'enable_webhook_notification',
+  ENABLE_MSTEAM_NOTIFICATION_IDENTIFIER = 'enable_msTeams_notification',
```
Comment (pankajsingh-harness) [inline src/modules/10-common/constants/Utils.ts:118]:
```suggestion
  ENABLE_MSTEAM_NOTIFICATION = 'enable_msTeams_notification',
```

File: src/modules/10-common/constants/Utils.ts [hunk 3]
```diff
@@ -112,7 +112,11 @@ export enum SettingType {
...
+  ENABLE_SLACK_NOTIFICATION_IDENTIFIER = 'enable_slack_notification'
```
Comment (pankajsingh-harness) [inline src/modules/10-common/constants/Utils.ts:119]:
```suggestion
  ENABLE_SLACK_NOTIFICATION = 'enable_slack_notification'
```

File: src/modules/27-platform/default-settings/interfaces/SettingType.types.ts [hunk 4]
```diff
@@ -12,7 +12,8 @@ export type SettingCategory = SettingDTO['category']
 export enum SettingGroups {
   test_group_2 = 'test_group_2',
   PERSPECTIVES_PREFERENCES = 'perspectives_preferences',
-  TICKETING_PREFERENCES = 'ticketing_preferences'
+  TICKETING_PREFERENCES = 'ticketing_preferences',
+  ENABLED_NOTIFICATION_CHANNELS_GROUP_IDENTIFIER = 'enabled_notification_channels'
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/interfaces/SettingType.types.ts:16]: NIT: Follow similar naming convention of have same key and value.
```suggestion
  ENABLED_NOTIFICATION_CHANNELS = 'enabled_notification_channels'
```

File: src/modules/27-platform/default-settings/RouteDestinations.tsx [hunk 5]
```diff
@@ -114,6 +136,17 @@ DefaultSettingsFactory.registerCategory('CE', {
   modulesWhereCategoryWillBeDisplayed: ['ce']
 })
 
+DefaultSettingsFactory.registerGroupHandler(SettingGroups.ENABLED_NOTIFICATION_CHANNELS_GROUP_IDENTIFIER, {
```
Comment (pankajsingh-harness) [inline src/modules/27-platform/default-settings/RouteDestinations.tsx:139]: NIT: Keep `registerGroupHandler` & `registerSettingHandler` together for better code readability. Move this above `registerSettingHandler` of this `groupId`

**Review by pankajsingh-harness — CHANGES_REQUESTED** [review]
NIT: Added some comments related to the naming convention.

**Review by pankajsingh-harness — DISMISSED** [review]

File: src/modules/10-common/constants/Utils.ts [hunk 6]
```diff
@@ -112,7 +112,11 @@ export enum SettingType {
   EXPORT_SERVICE_VARIABLES_AS_ENV_VARIABLES = 'export_service_variables_as_env_variables',
   TICKETING_TOOL = 'ticketing_tool',
   TICKETING_TOOL_CONNECTOR = 'ticketing_tool_connector',
-  AIDA = 'aida'
+  AIDA = 'aida',
+  ENABLE_PAGERDUTY_NOTIFICATION = 'enable_pagerduty_notification',
+  ENABLE_WEBHOOK_NOTIFICATION = 'enable_webhook_notification',
+  ENABLE_MSTEAM_NOTIFICATION = 'enable_msTeams_notification',
```
Comment (mayankVermaHarness) [inline src/modules/10-common/constants/Utils.ts:118]:
```suggestion
  ENABLE_MSTEAMS_NOTIFICATION = 'enable_msTeams_notification',
```

Comment (namanharness) [reply src/modules/10-common/constants/Utils.ts]: done

**Review by mayankVermaHarness — APPROVED** [review]

**Review by Eric761 — APPROVED** [review]

**Pull request merged**